RedHatInsights/rhsm-subscriptions
# ENT-3341, ENT-3744: Implement MarketplaceSubscriptionIdCollector

# Testing
* Make sure there are offerings in the offering table that matches our SKUs of interest.  See JIRA ticket comments for insert statements for testing
* Find an account to use that has subscriptions (https://webadmin.corp.qa.redhat.com/subscription-admin).  See JIRA ticket comments for account numbers I used
* Get OCP or OSD events into your table for the associated account number
* Make sure the account_config table has your account opted in
* Kick off an hourly tally for a time frame that encompasses the events in your event table

Assuming that your subscriptions table was empty to start with, after this logic executes you will see an entry in the rhsm-subscription.subscription table that matches a subscription fetched from the subscription service
```bash
curl --cert-type P12 --cert nonprod-insights.jks:"$(cat jks_pass)" 'https://subscription.qa.api.redhat.com/svcrest/subscription/v5/search/criteria;oracle_account_number=account123/options;products=ALL;showExternalReferences=true'
```

If you were to execute this again, without clearing your subscription table, the logic shouldn't execute a GET request to the subscription service this time around, because that data is in the swatch database now.


# app properties tips

```yaml
# these settings currently reside in the application-worker.yaml file
rhsm-subscriptions:
  subscription:
    useStub: false
    url: https://subscription.qa.api.redhat.com/svcrest/subscription/v5
    keystore: nonprod-insights.jks
    keystorePassword:
# turns on logging for HTTP events to help verify when the subscription service is/isn't getting called
logging:
  level:
    org:
      apache:
        http: DEBUG
```
